config/Cargo.toml

```diff
@@ -0,0 +1,16 @@
+[package]
+name = "cosmic-settings-config"
+version = "0.1.0"
+edition = "2021"
+
+[features]
+smithay = ["dep:smithay"]
+
+[dependencies]
+cosmic-config = { git = "https://github.com/pop-os/libcosmic" }
+serde = { version = "1.0", features = ["derive"] }
+serde_with = "3.8.1"
+smithay = { git = "https://github.com/smithay/smithay.git", optional = true }
```

Comment:
So the idea is, that the executable using this (e.g. cosmic-comp) specifies the commit and features?

Does that even work, can we disable `default-features`, if `cosmic-settings-config` depends on smithay with a default feature set? I would guess not, because features are supposed to be purely additive. So this probably implicitly re-enables `use_system_lib` for example.

Smithay is quite a big and heavy dependency and the only uses I am seeing here is `ModifiersState` and some conversions. Lets just copy `ModifiersState` over and move the conversions into https://github.com/pop-os/cosmic-comp/pull/400.

That means we loose the nice `From`-implementations because of the orphan rule, but those can just be plain functions and we can still implement all the serde-stuff with `serde(foreign)`, if necessary.

Reply:
It's not a default feature, so it's not required to disable anything. Only cosmic-comp is opting into this for the From/Into and PartialEq trait implementations right now.

Reply:
Right, but this would enable any default features, since `cosmic-settings-config` doesn't declare `default-features = false`. But that isn't the point, but rather how fragile this is, just for a couple of `From/Into`-impls.